### PR TITLE
chore: backport fixes into v1.x for rc5

### PR DESCRIPTION
**Pull request by cmwaters**
This PR backports the following commits in preparation for rc5

- db3e6584 feat!: run min gas pfb decorator in process proposal (#1985)
- 43b933ee fix!: implement full antehandler checks (#1984)
- c7281149 feat: implement MinGasPFBDecorator (#1936)
- d34f88e0 feat: introduce ibc redundant relay decorator (#1934)
- 639d6243 feat: use panics and recovery appropriately (#1948)
- cd212def fix!: signature validity checks in ProcessProposal (#1981)
- a27d2520 fix!: add namespace version check in pfbs (#1983)

Ideally we should MERGE COMMIT and not SQUASH COMMIT so we can preserve the commit history

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: x/blob/ante/ante.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+// AnteHandle implemnts the AnteHandler interface. It checks to see
+// if the transaction contains a MsgPayForBlobs and if so, checks that
+// the transaction has allocated enough gas.
+//
+// TODO: We need to run this antehandler in process proposal
+// however this will be a breaking change.
+func (d MinGasPFBDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	if !ctx.IsCheckTx() {
+		return next(ctx, tx, simulate)
+	}
```
Comment (evan-forbes) [inline x/blob/ante/ante.go:33]: no #1985 ?

Comment (evan-forbes) [reply x/blob/ante/ante.go:33]: https://github.com/celestiaorg/celestia-app/pull/1985#pullrequestreview-1495014460

Comment (cmwaters) [reply x/blob/ante/ante.go:33]: Will add it now

**Comment by evan-forbes** [comment]
we also have https://github.com/celestiaorg/celestia-app/pull/1851#pullrequestreview-1494058501, but as stated there, while its not an unplanned validity rule, it is an arguably not necessary change to the state machine. Definitley not blocking on merging that, I think we should merge this as soon as possible and if we don't get #1851, then we will just have to deal with overwriting the genesis state forever, which is nbd

**Review by evan-forbes — APPROVED** [review]

**Review by SweeXordious — APPROVED** [review]

**Pull request merged**